crypto/pkcs12/pbkdf.go

```diff
@@ -0,0 +1,214 @@
+// Entire File adapted from https://github.com/Azure/go-pkcs12/blob/master/pbkdf.go
+// and https://github.com/Azure/go-pkcs12/blob/master/crypto.go under the MIT
+// license.  These functions implement cryptographic processes used in decoding
+// password protected pfx objects adhering to the pkcs12 standard.
+package pkcs12
+
+import (
+	"errors"
+	"math/big"
+	"unicode/utf16"
+	"unicode/utf8"
+)
+
+var (
+	deriveKeyByAlg = map[string]func(salt, password []byte, iterations int) []byte{
+		pbeWithSHAAnd3_KeyTripleDES_CBC: func(salt, password []byte, iterations int) []byte {
+			return pbkdf(sha1Sum, 20, 64, salt, password, iterations, 1, 24)
+		},
+		pbewithSHAAnd40BitRC2CBC: func(salt, password []byte, iterations int) []byte {
+			return pbkdf(sha1Sum, 20, 64, salt, password, iterations, 1, 5)
+		},
+	}
+	deriveIVByAlg = map[string]func(salt, password []byte, iterations int) []byte{
+		pbeWithSHAAnd3_KeyTripleDES_CBC: func(salt, password []byte, iterations int) []byte {
+			return pbkdf(sha1Sum, 20, 64, salt, password, iterations, 2, 8)
+		},
+		pbewithSHAAnd40BitRC2CBC: func(salt, password []byte, iterations int) []byte {
+			return pbkdf(sha1Sum, 20, 64, salt, password, iterations, 2, 8)
+		},
+	}
+)
+
+func pbkdf(hash func([]byte) []byte, u, v int, salt, password []byte, r int, ID byte, size int) (key []byte) {
```

Comment:
This could eventually be submitted to golang.org/x/crypto, but I doubt they would take it since PBKDF2 is the current standard.